tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,299 @@
+
+    Return:
+        This function will return True if all critical processes are running and
+        all BGP sessions are established. Otherwise it will return False.
+    """
+
+    return check_all_critical_processes_status(duthost) and duthost.check_bgp_session_state(up_bgp_neighbors, "established")
+
+
+def postcheck_critical_processes_status(duthost, up_bgp_neighbors):
+    """Calls the functions to post-check the status of critical processes and
+       state of BGP sessions.
+
+    Args:
+        duthost: Host DUT.
+        skip_containers: A list contains the container names which should be skipped.
+
+    Return:
+        If all critical processes are running and all BGP sessions are established, it
+        returns True. Otherwise it will call the function to do post-check every 30 seconds
+        for 3 minutes. It will return False after timeout
+    """
+    return wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS,
+                      post_test_check, duthost, up_bgp_neighbors)
+
+
+def stop_containers(duthost, container_autorestart_states, skip_containers):
+    """Stops the running containers and returns their names .
+
+    Args:
+        duthost: Host DUT.
+        container_autorestart_states: A dictionary which key is container name and
+        value is the state of autorestart feature.
+        skip_containers: A list contains the container names which should be skipped.
+
+    Return:
+        A list which contains the container names which are stopped to run.
+    """
+    stopped_containers_list = []
+
+    for container_name in container_autorestart_states.keys():
+        if container_name not in skip_containers:
+            logger.info("Stopping the container '{}'...".format(container_name))
+            duthost.shell("sudo systemctl stop {}.service".format(container_name))
+            logger.info("Waiting until container '{}' is stopped...".format(container_name))
+            stopped = wait_until(CONTAINER_STOP_THRESHOLD_SECS,
+                                 CONTAINER_CHECK_INTERVAL_SECS,
+                                 check_container_state, duthost, container_name, False)
+            pytest_assert(stopped, "Failed to stop container '{}'".format(container_name))
+            logger.info("Container '{}' was stopped".format(container_name))
+            stopped_containers_list.append(container_name)
+
+    return stopped_containers_list
+
+
+def check_alerting_message(duthost, stopped_containers_list):
+    """Checks whether the names of stopped containers appear in Monit alerting message.
+
+    Args:
+        duthost: Host DUT.
+        stopped_containers_list: A list of stopped container names.
+
+    Return:
+        None.
+    """
+    alerting_message = duthost.shell("sudo cat /var/log/syslog | grep -m 1 '.*monit.*container_checker'",
+                                     module_ignore_errors=True)
+
+    pytest_assert(len(alerting_message["stdout_lines"]) > 0,
```

Comment:
```suggestion
    pytest_assert(len(stopped_containers_list) == 0 or len(alerting_message["stdout_lines"]) > 0,
```
Although it's not likely to happen, I think it's better to add this check.

Reply:
Great suggestion! Will update.

Reply:
Updated!